core/src/main/java/org/testcontainers/containers/VncRecordingContainer.java

```diff
@@ -97,18 +106,44 @@ public void saveRecordingToFile(File file) {
 
     @SneakyThrows
     public InputStream streamRecording() {
-        reencodeRecording();
+        String newRecordingFileName = recordingFormat.reencodeRecording(this, ORIGINAL_RECORDING_FILE_NAME);
 
         TarArchiveInputStream archiveInputStream = new TarArchiveInputStream(
-                dockerClient.copyArchiveFromContainerCmd(getContainerId(), RECORDING_FILE_NAME).exec()
+                dockerClient.copyArchiveFromContainerCmd(getContainerId(), newRecordingFileName).exec()
         );
         archiveInputStream.getNextEntry();
         return archiveInputStream;
     }
 
-    private void reencodeRecording() throws IOException, InterruptedException {
-        String newFlvOutput = "/newScreen.flv";
-        execInContainer("ffmpeg" , "-i", RECORDING_FILE_NAME, "-vcodec", "libx264", newFlvOutput);
-        execInContainer("mv" , "-f", newFlvOutput, RECORDING_FILE_NAME);
+
+    public enum VncRecordingFormat {
+        FLV("flv") {
+            @Override
+            String reencodeRecording(VncRecordingContainer container, String source) throws IOException, InterruptedException {
```

Comment:
What about `@NonNull` for the parameters

Reply:
Done, Thanks